### ISSUE TITLE
Formatted number with currencies  are not right alligned on tables

### ISSUE THREAD

**Issue by rephus** [lightdash/lightdash#2062]
### Describe the bug

Not sure if a regression.  But these numbers (like percents) shoudl be alligned to the right 

![image](https://user-images.githubusercontent.com/1983672/167789628-b1847ed7-4691-4f01-b3a8-2a382d6bdd1c.png)

### Expected behavior

All currencies and numbers should be alligned on the right

**Comment by ZeRego** [comment]
@rephus @TuringLovesDeathMetal 

Currently we align based on the value. There is a `isNan` function that determines if the value is not a number. The problem is that we need to check against the formatted value instead of the raw value. 

I think we should instead check if the field type is numeric and stop relying on values. The only exception for this is the table calculations where there is no specific type, I suggest we treat it as a number and right align.

This relates to another point where we allow string dimensions/metrics to be formatted like a number. I believe we should treat the values per dimension/metric type instead of being smart since this is going to bring to many edge cases. If the user has a string dimension that he want's to use as a number, they can mark the type as number and use the `sql` field for parse the string into number.

**Comment by ZeRego** [comment]
For scenario 3 what is the metric type ? because that would be the decider. 
Imagine you have type `SUM`, independently of what your custom SQL looks like we will treat it as a numeric value and right align it. It the case you mentioned they should define the metric type as `STRING`

**Comment by rephus** [comment]
I agree it looks terrible, but this is how google/excel do it... so it is kind of accepted.. 

![image](https://user-images.githubusercontent.com/1983672/167857023-62cb9b56-dd7b-4fd5-bbc7-79550fa58445.png)
